library/proc_macro/src/lib.rs

```diff
@@ -1351,6 +1352,15 @@ impl Literal {
         Literal::new(bridge::LitKind::ByteStr, &string, None)
     }
 
+    /// C string literal.
+    #[unstable(feature = "proc_macro_c_str_literals", issue = "119750")]
+    pub fn c_string(string: &CStr) -> Literal {
+        let quoted = format!("{:?}", string);
```

Comment:
relying on the `Debug` impl doesnt sound very good, can we instead `to_bytes()` (I forgot whether that contains the null byte so you may need to strip that) and then `escape_ascii()` like `byte_string` above?

Reply:
Ah yeah that is a lot nicer isn't it...thanks! I wasn't sure either -- turns out `to_bytes` does not include the NUL.